R/VaribleChartsSubgroups.R

```diff
@@ -1,47 +1,46 @@
+      dataset.factors <- .readDataSetToEnd(columns=variables, columns.as.factor=time)
+    } else {
+      dataset         <- .readDataSetToEnd(columns.as.numeric=variables)
+      dataset.factors <- .readDataSetToEnd(columns=variables)
+    }
+  }
+  if (makeTime && length(variables) > 0) {
+    # For the time-variable we first convert the original factor to a character so that the order of input is kept!
+    dataset[[.v(time)]] <- as.character(dataset[[.v(time)]])
+  }
   #Checking for errors in the dataset
   .hasErrors(dataset, type = c('observations', 'infinity', 'missingValues'),
              all.target = options$variables,
              observations.amount = c(' < 2'), exitAnalysisIfErrors = TRUE)
 
-  jaspResults[["intro"]] <- createJaspHtml(gettext("Select one of the control charts from the interface."))
-  jaspResults[["intro"]]$position <- 0
 #X bar chart
   if(options$Xbarchart && is.null(jaspResults[["Xbarchart"]]) &&  length(options$variables) > 1){
-    jaspResults[["XbarPlot"]] <- createJaspPlot(title =  gettext("X bar chart"), width = 700, height = 350)
+    jaspResults[["XbarPlot"]] <- createJaspPlot(title =  gettext("X-bar & R Control Chart"), width = 700, height = 350)
     jaspResults[["XbarPlot"]]$dependOn(c("Xbarchart", "variables"))
     XbarPlot <- jaspResults[["XbarPlot"]]
-    XbarPlot$plotObject <- .XbarchartNoId(dataset = dataset[, options$variables], options = options)
-
-    jaspResults[["RPlot"]] <- createJaspPlot(title =  gettext("R chart"), width = 700, height = 350)
-    jaspResults[["RPlot"]]$dependOn(c("Xbarchart", "variables"))
-    RPlot<- jaspResults[["RPlot"]]
-    RPlot$plotObject <- .RchartNoId(dataset = dataset[, options$variables], options = options)
+    XbarPlot$plotObject <- jaspGraphs::ggMatrixPlot(plotList = list(.RchartNoId(dataset = dataset, options = options, time = ifelse(makeTime, TRUE, FALSE)), .XbarchartNoId(dataset = dataset, options = options, time = ifelse(makeTime, TRUE, FALSE))), layout = matrix(1:2, 2), removeXYlabels= "x")
   }
 
 #S Chart
   if(options$Schart && is.null(jaspResults[["Schart"]]) &&  length(options$variables) > 1){
-    jaspResults[["XbarPlot"]] <- createJaspPlot(title =  gettext("X bar chart"), width = 700, height = 350)
-    jaspResults[["XbarPlot"]]$dependOn(c("Schart", "variables"))
-    XbarPlot <- jaspResults[["XbarPlot"]]
-    XbarPlot$plotObject <- .XbarchartNoId(dataset = dataset[, options$variables], options = options)
-
-    jaspResults[["SPlot"]] <- createJaspPlot(title = gettext("S Chart"), width = 700, height = 350)
+    jaspResults[["SPlot"]] <- createJaspPlot(title = gettext("Xbar & s Control Chart"), width = 700, height = 350)
     jaspResults[["SPlot"]]$dependOn(c("Schart", "variables"))
     SPlot<- jaspResults[["SPlot"]]
-    SPlot$plotObject <- .Schart(dataset = dataset, options = options)
+    SPlot$plotObject <- .XbarSchart(dataset = dataset, options = options, time = ifelse(makeTime, TRUE, FALSE))
```

Comment:
```suggestion
    SPlot$plotObject <- .XbarSchart(dataset = dataset, options = options, time = makeTime)
```
if it's a boolean already just use it directly?